### PR TITLE
Document events API

### LINKED ISSUE #2632: Document Events API
#### Summary

I think it would be good to document the events API.

#### Why do we need this ?

Because we expect people to build integrations on top of it. Think gateway uptime monitoring, airtime tracking, audit logging, etc.

#### What is already there? What do you see now?

We define events througout the codebase and we have a powerful API to subscribe to events.

#### What is missing? What do you want to see?

Documentation for the Events API in the API reference. 

#### How do you propose to document this?

- Document the gRPC service similar to how we document the other gRPC services
- Document how it maps to HTTP (with server-sent events)
- Up for discussion: Add generated documentation for the events defined by The Things Stack

### PR DESCRIPTION
**Pull request by htdvisser**
#### Summary

This adds documentation for our events API.

Closes #2632

#### Changes

- Some more changes to `pkg/events` in addition to #3018
- Generate data file with defined events
- Document the Events API in the proto file
- Add a documentation page for the Events API
- Add some data types to definitions

#### Testing

Screen Shot 2020-08-12 at 13 56 21

Screen Shot 2020-08-12 at 13 59 41

#### Notes for Reviewers

- Everyone, please approve the small changes in code-owned files for a green merge button
- @benolayinka please review changes in `/doc`
- @rvolosatovs please review changes in `/pkg/events`

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [x] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Review by KrishnaIyer — APPROVED** [review]
GS LGTM

**Review by adriansmares — APPROVED** [review]
LGTM. This stabilization is also `CHANGELOG.md` worthy imo.

**Review by benolayinka — APPROVED** [review]
Sentence case for Event descriptions

Screenshot 2020-08-13 at 16 01 20

**Review by johanstokking — APPROVED** [review]
Code owned files LGTM

**Comment by htdvisser** [comment]
> @benolayinka: Sentence case for Event descriptions

I think that might become a bit nasty. Hugo doesn't have something builtin for uppercasing the first character, so it would be a combination of uppercasing a substring, and then appending another substring. Is this important enough to do such nasty things?

**Comment by benolayinka** [comment]
> > @benolayinka: Sentence case for Event descriptions
> 
> I think that might become a bit nasty. Hugo doesn't have something builtin for uppercasing the first character, so it would be a combination of uppercasing a substring, and then appending another substring. Is this important enough to do such nasty things?

Nah, no big deal

**Comment by kschiffer** [comment]
```css
.event-descriptions:first-letter {
    text-transform: uppercase;
}
```

Can that be of help?

**Review by rvolosatovs — APPROVED** [review]

File: pkg/networkserver/observability.go [hunk 1]
```diff
@@ -106,6 +106,7 @@ var (
 	evtDropDataUplink = events.Define(
```
Comment (rvolosatovs) [inline pkg/networkserver/observability.go:112]: There's a bunch of typing left here, e.g. for MAC events - please file an issue and I will address it.

Comment (htdvisser) [reply pkg/networkserver/observability.go:112]: I added data types for most (if not all) of the NS events. I did notice that the following seem to be unused:

- `evtEnqueueProprietaryMACAnswer`
- `evtEnqueueProprietaryMACRequest`
- `evtReceiveProprietaryMAC`

**Pull request merged**